worlds/zork_grand_inquisitor/world.py

```diff
@@ -0,0 +1,208 @@
+                data: ZorkGrandInquisitorLocationData = location_data[location_enum_item]
+
+                location: ZorkGrandInquisitorLocation = ZorkGrandInquisitorLocation(
+                    self.player,
+                    location_enum_item.value,
+                    data.archipelago_id,
+                    region_mapping[data.region],
+                )
+
+                location.event = isinstance(location_enum_item, ZorkGrandInquisitorEvents)
+
+                if location.event:
+                    location.place_locked_item(
+                        ZorkGrandInquisitorItem(
+                            data.event_item_name,
+                            ItemClassification.progression,
+                            None,
+                            self.player,
+                        )
+                    )
+
+                location_access_rule: str = location_access_rule_for(location_enum_item, self.player)
+
+                if location_access_rule != "lambda state: True":
+                    location.access_rule = eval(location_access_rule)
+
+                region.locations.append(location)
+
+            # Connections
+            region_exit: ZorkGrandInquisitorRegions
+            for region_exit in region_data[region_enum_item].exits or tuple():
+                entrance_access_rule: str = entrance_access_rule_for(region_enum_item, region_exit, self.player)
+
+                if entrance_access_rule == "lambda state: True":
+                    region.connect(region_mapping[region_exit])
+                else:
+                    region.connect(region_mapping[region_exit], rule=eval(entrance_access_rule))
+
+            self.multiworld.regions.append(region)
+
+    def create_items(self) -> None:
+        quick_port_foozle: bool = bool(self.options.quick_port_foozle)
+        start_with_hotspot_items: bool = bool(self.options.start_with_hotspot_items)
+
+        item_pool: List[ZorkGrandInquisitorItem] = list()
+
+        item: ZorkGrandInquisitorItems
+        data: ZorkGrandInquisitorItemData
+        for item, data in item_data.items():
+            tags: Tuple[ZorkGrandInquisitorTags, ...] = data.tags or tuple()
+
+            if ZorkGrandInquisitorTags.FILLER in tags:
+                continue
+            elif ZorkGrandInquisitorTags.HOTSPOT in tags and start_with_hotspot_items:
+                continue
+
+            item_pool.append(self.create_item(item.value))
+
+        total_locations: int = len(self.multiworld.get_unfilled_locations(self.player))
+        item_pool += [self.create_filler() for _ in range(total_locations - len(item_pool))]
+
+        self.multiworld.itempool += item_pool
+
+        if quick_port_foozle:
+            self.multiworld.early_items[self.player][ZorkGrandInquisitorItems.ROPE.value] = 1
+            self.multiworld.early_items[self.player][ZorkGrandInquisitorItems.LANTERN.value] = 1
+
+            if start_with_hotspot_items is False:
```

Comment:
I gave my reasoning [here](https://github.com/ArchipelagoMW/Archipelago/pull/2539#discussion_r1519692903) in case you are curious. That being said, the distinction is not really useful when you aren't dealing with mixed types, so I changed all usage of `is True|False`